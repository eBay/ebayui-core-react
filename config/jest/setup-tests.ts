import '@testing-library/jest-dom'

declare var global: typeof globalThis;

if (typeof window.URL.createObjectURL === 'undefined') {
  Object.defineProperty(window.URL, 'createObjectURL', { value: () => {} });
}
