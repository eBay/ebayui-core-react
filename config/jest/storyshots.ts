import { render } from "@testing-library/react";
import initStoryshotsAddon from "@storybook/addon-storyshots";
import type { StoryshotsOptions } from "@storybook/addon-storyshots/dist/ts3.9/api/StoryshotsOptions";

const reactTestingLibrarySerializer = {
    print: (val, serialize, indent) => serialize(val.asFragment()),
    test: (val) => val && val.hasOwnProperty("container"),
};

export function initStoryshots(config: StoryshotsOptions) {
    initStoryshotsAddon({
        ...config,
        renderer: render,
        snapshotSerializers: [reactTestingLibrarySerializer],
    });
}
