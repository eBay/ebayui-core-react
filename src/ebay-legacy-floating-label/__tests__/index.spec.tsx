import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { EbayLegacyFloatingLabel } from '../../index'
import { initStoryshots } from '../../../config/jest/storyshots'

describe('<EbayLegacyFloatingLabel>', () => {
    describe('on input change', () => {
        it('should fire an event', () => {
            const spy = jest.fn()
            const { getByLabelText } = render(
                <EbayLegacyFloatingLabel id="input" onChange={spy} label="look ma, I float" />
            )
            const input = getByLabelText('look ma, I float')
            userEvent.type(input, '123')

            expect(spy).toBeCalled()
        })
    })

    describe('on input blur', () => {
        it('should fire an event', () => {
            const spy = jest.fn()
            const { getByLabelText } = render(
                <EbayLegacyFloatingLabel id="input" onBlur={spy} label="look ma, I float" />
            )
            const input = getByLabelText('look ma, I float')
            fireEvent.blur(input)

            expect(spy).toBeCalled()
        })
    })

    describe('on input focus', () => {
        it('should fire an event', () => {
            const spy = jest.fn()
            const { getByLabelText } = render(
                <EbayLegacyFloatingLabel id="input" onFocus={spy} label="look ma, I float" />
            )

            const input = getByLabelText('look ma, I float')
            fireEvent.focus(input)

            expect(spy).toBeCalled()
        })
    })

    describe('on passing a ref', () => {
        it('should set the ref as the input reference', () => {
            const ref = React.createRef() as any
            render(<EbayLegacyFloatingLabel ref={ref} label="look ma, I float" />)

            expect(ref.current.tagName).toBe('INPUT')
        })
    })
})

global.getComputedStyle = jest.fn().mockReturnValue({ backgroundColor: 'rgb(255, 255, 255)' })

initStoryshots({
    config: ({ configure }) =>
        configure(() => {
            require('./index.stories')
        }, module)
})
