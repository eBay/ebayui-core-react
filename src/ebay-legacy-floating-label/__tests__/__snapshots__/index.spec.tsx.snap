// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-legacy-floating-label Autofocus test 1`] = `
<DocumentFragment>
  <div>
    <span
      class="legacy-floating-label"
    >
      <label
        class="legacy-floating-label__label legacy-floating-label__label--animate"
      >
        Autofocused field
      </label>
      <span
        class="textbox"
      >
        <input
          class="textbox__control legacy-textbox-underline"
          type="text"
          value=""
        />
      </span>
    </span>
    <br />
    <span
      class="legacy-floating-label"
    >
      <label
        class="legacy-floating-label__label legacy-floating-label__label--inline"
      >
        Regular field
      </label>
      <span
        class="textbox"
      >
        <input
          class="textbox__control legacy-textbox-underline"
          type="text"
          value=""
        />
      </span>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-legacy-floating-label Control value from outside 1`] = `
<DocumentFragment>
  <p>
    <span
      class="legacy-floating-label"
    >
      <label
        class="legacy-floating-label__label legacy-floating-label__label--inline"
      >
        Will convert to lowercase
      </label>
      <span
        class="textbox"
      >
        <input
          class="textbox__control legacy-textbox-underline"
          type="text"
          value=""
        />
      </span>
    </span>
  </p>
  <p>
    <button
      class="btn btn--secondary"
    >
      Change text
    </button>
  </p>
  <p>
    <button
      class="btn btn--secondary"
    >
      Clear
    </button>
  </p>
</DocumentFragment>
`;

exports[`Storyshots ebay-legacy-floating-label Default floating label 1`] = `
<DocumentFragment>
  <form>
    <p>
      <span
        class="legacy-floating-label"
      >
        <label
          class="legacy-floating-label__label legacy-floating-label__label--inline"
          for="input-1"
        >
          Email address
        </label>
        <span
          class="textbox"
        >
          <input
            class="textbox__control legacy-textbox-underline"
            id="input-1"
            type="email"
            value=""
          />
        </span>
      </span>
    </p>
    <p>
      <button
        class="btn btn--secondary"
        type="submit"
      >
        Submit
      </button>
    </p>
  </form>
</DocumentFragment>
`;

exports[`Storyshots ebay-legacy-floating-label Floating label disabled 1`] = `
<DocumentFragment>
  <span
    class="legacy-floating-label"
  >
    <label
      class="legacy-floating-label__label legacy-floating-label__label--disabled legacy-floating-label__label--inline"
      for="input-4"
    >
      Email address
    </label>
    <span
      class="textbox"
    >
      <input
        class="textbox__control legacy-textbox-underline"
        disabled=""
        id="input-4"
        type="text"
        value=""
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-legacy-floating-label Floating label invalid 1`] = `
<DocumentFragment>
  <span
    class="legacy-floating-label"
  >
    <label
      class="legacy-floating-label__label legacy-floating-label__label--inline"
      for="input-5"
    >
      Email address
    </label>
    <span
      class="textbox"
    >
      <input
        aria-invalid="true"
        class="textbox__control legacy-textbox-underline"
        id="input-5"
        type="text"
        value=""
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-legacy-floating-label Floating label with controlled value 1`] = `
<DocumentFragment>
  <span
    class="legacy-floating-label"
  >
    <label
      class="legacy-floating-label__label"
      for="input-21"
    >
      Email address
    </label>
    <span
      class="textbox"
    >
      <input
        class="textbox__control legacy-textbox-underline"
        id="input-21"
        type="text"
        value="can't change"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-legacy-floating-label Floating label with placeholder 1`] = `
<DocumentFragment>
  <span
    class="legacy-floating-label"
  >
    <label
      class="legacy-floating-label__label legacy-floating-label__label--inline"
      for="input-3"
    >
      Email address
    </label>
    <span
      class="textbox"
    >
      <input
        class="textbox__control legacy-textbox-underline"
        id="input-3"
        placeholder="placeholder text"
        type="text"
        value=""
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-legacy-floating-label Floating label with value 1`] = `
<DocumentFragment>
  <span
    class="legacy-floating-label"
  >
    <label
      class="legacy-floating-label__label"
      for="input-2"
    >
      Email address
    </label>
    <span
      class="textbox"
    >
      <input
        class="textbox__control legacy-textbox-underline"
        id="input-2"
        type="text"
        value="test"
      />
    </span>
  </span>
</DocumentFragment>
`;
